src/common/file_location.c

```diff
@@ -109,93 +171,115 @@ char *dt_loc_find_install_dir(const char *suffix, const char *searchname)
-  g_free(suffix);
-  darktable.plugindir = dt_loc_init_generic(plugindir, directory ? directory : DARKTABLE_LIBDIR);
+#if defined(__APPLE__)
+  char *directory = dt_loc_find_install_dir(DARKTABLE_LIBDIR);
+  darktable.plugindir = dt_loc_init_generic(plugindir, application_directory, directory ? directory : DARKTABLE_LIBDIR);
   g_free(directory);
 #else
-  darktable.plugindir = dt_loc_init_generic(plugindir, DARKTABLE_LIBDIR);
+  darktable.plugindir = dt_loc_init_generic(plugindir, application_directory, DARKTABLE_LIBDIR);
 #endif
+  dt_check_opendir("darktable.plugindir", darktable.plugindir);
 }
 
-void dt_loc_init_localedir(const char *localedir)
+void dt_check_opendir(const char* text, const char* directory)
 {
-#if defined(__APPLE__) || defined(_WIN32)
-  char *suffix = g_build_filename("share", "locale", NULL);
-  char *directory = dt_loc_find_install_dir(suffix, darktable.progname);
-  g_free(suffix);
-  darktable.localedir = dt_loc_init_generic(localedir, directory ? directory : DARKTABLE_LOCALEDIR);
-#ifdef __APPLE__
-  if(directory && !localedir) //bind to bundle path
-    bindtextdomain(GETTEXT_PACKAGE, darktable.localedir);
+  if (!directory)
+  {
+    fprintf(stderr, "directory for %s has not been set.\n", text);
+    exit(EXIT_FAILURE);
+  } 
+
+#if _WIN32
+  DWORD attribs = GetFileAttributesA(directory);
+  if (attribs != INVALID_FILE_ATTRIBUTES &&
+      (attribs & FILE_ATTRIBUTE_DIRECTORY))
+  {
+    dt_print(DT_DEBUG_DEV, "%s: %s\n", text, directory);
+  }
+  else
+  {
+    fprintf(stderr, "%s: directory '%s' fails to open.'\n", text, directory);
+    exit(EXIT_FAILURE);
+  }
+#else
+  DIR* dir = opendir(directory);
+  if (dir)
+  {
+    dt_print(DT_DEBUG_DEV, "%s: %s\n", text, directory);
+    closedir(dir);
+  } 
+  else 
+  {
+    fprintf(stderr, "opendir '%s' fails with: '%s'\n", directory, strerror(errno));
+    exit(EXIT_FAILURE);
+  }
+#endif
+}
+
+void dt_loc_init_localedir(const char* application_directory, const char *localedir)
+{
+#if defined(__APPLE__)
+  char *directory = dt_loc_find_install_dir(DARKTABLE_LOCALEDIR);
+  darktable.localedir = dt_loc_init_generic(localedir, application_directory, directory ? directory : DARKTABLE_LOCALEDIR);
+  g_free(directory);
+#else
+  darktable.localedir = dt_loc_init_generic(localedir, application_directory, DARKTABLE_LOCALEDIR);
 #endif
+  dt_check_opendir("darktable.localedir", darktable.localedir);
+}
+
+void dt_loc_init_datadir(const char* application_directory, const char *datadir)
+{
+#if defined(__APPLE__)
+  char *directory = dt_loc_find_install_dir(DARKTABLE_DATADIR);
+  darktable.datadir = dt_loc_init_generic(datadir, application_directory, directory ? directory : DARKTABLE_DATADIR);
   g_free(directory);
 #else
-  darktable.localedir = dt_loc_init_generic(localedir, DARKTABLE_LOCALEDIR);
+  darktable.datadir = dt_loc_init_generic(datadir, application_directory, DARKTABLE_DATADIR);
 #endif
+  dt_check_opendir("darktable.datadir", darktable.datadir);
```

Comment:
I get an error:
```
$ /Volumes/darktable/darktable.app/Contents/MacOS/darktable -d dev
0.001000 application_directory: /Volumes/darktable/darktable.app/Contents/MacOS
path lookup '/Volumes/darktable/darktable.app/Contents/MacOS//Volumes/darktable/darktable.app/Contents/Resources/../share/darktable' fails with: 'No such file or directory'
```

Reply:
I was not yet able to build the mac image because of the scripts python 2 state being incompatible to a standard macs python 3 installation. So I had to patch gtk-mac-bundler to use python2 instead of python 3 like so:
```
head ~/.local/bin/gtk-mac-bundler
#!/usr/bin/env python2
```

I can confirm this issue when trying to launch the mac image (/Volumes/darktable/darktable.app/Contents/Resources/share/darktable) or the installed application coming from the mac image (/Applications/darktable.app/Contents/MacOS/darktable).
Since the binary tree and install dir version work fine I guess the error happens when creating of the mac image.

Reply:
The bundle path handling shall be implemented and working now